parquet_derive/src/lib.rs

```diff
@@ -0,0 +1,294 @@
+#![feature(impl_trait_in_bindings, box_patterns)]
+#![recursion_limit = "128"]
+
+extern crate proc_macro;
+extern crate proc_macro2;
+extern crate syn;
+#[macro_use]
+extern crate quote;
+
+extern crate parquet;
+
+#[allow(unused_imports)]
+use syn::{
+  parse_macro_input, AngleBracketedGenericArguments, Data, DataStruct, DeriveInput,
+  Field, Fields, Generics, Ident, Lifetime, ParenthesizedGenericArguments, Path,
+  PathArguments, Type, TypePath,
+};
+
+#[derive(Debug)]
+struct FieldInfo {
+  field_name: Ident,
+  field_lifetime: Option<Lifetime>,
+  field_type: Ident,
+  is_option: bool,
+  column_writer_variant: proc_macro2::TokenStream,
```

Comment:
any reason we need to depend both on proc_macro and proc_macro2? just curious :)

Reply:
I chose `proc_macro2` to maximize our backwards compatibility, from their description:

```
A small shim over the proc_macro crate in the compiler intended to multiplex the stable interface as of 1.15.0 and the interface as of 1.30.0.
```

So I take in a compiler-defined `proc_macro::TokenStream` and convert it to `proc_macro2::TokenStream` so I can work on it in a rustc version-independent way. See as we only claim compatibility with nightly I can see this being unnecessary.

I'm open to removing it, I think it would just a matter of renaming to `proc_macro::TokenStream`, etc.